### ISSUE TITLE
Cannot select multiple recipients upon completion

### ISSUE THREAD

**Issue by the-x-at** [neomutt/neomutt#3504]
The first step in composing a new email message is to choose the recipient(s). During this step, when auto-completing addresses, selecting multiple addresses in the query menu returns only a single address, which is the one the cursor is currently on.

## Expected Behavior

All selected recipients should be chosen and added as a comma separated list to the entry field.

## Actual Behavior

Only the email address where the cursor is located will get into the entry field.

## Steps to Reproduce

I usually work with address completion via `abook`, that was when I discovered this bug. But it appears the query dialog is somehow broken. I was able to create a minimal config file just using aliases, `minimal.muttrc`, that is already able to reproduce the bug:

    alias beatrix  Beatrix Jones <bj@example.com>
    alias beatrice Beatrice Williams <bw@example.com>
    alias bob      Bob Cramer <bc@example.com>

Starting NeoMutt with `neomutt -F minimal.muttrc` and then hitting the `m` key to compose a new email presents you with the `To:` address recipient field. Entering `bea` will complete till `beatri` and then present a menu with two entries where we select both of them hitting the `t` key:

    q:Exit  d:Del  u:Undel  o:Sort  O:Rev-Sort  <Enter>:Select  ?:Help
      1  * beatrice        Beatrice Williams <bw@example.com>                       |
      2  * beatrix         Beatrix Jones <bj@example.com>                           |

Let's put the cursor on entry number two and hit `<enter>`. The `To:` field now contains only 

    To: Beatrix Jones <bj@example.com>

Furthermore, if we just compose an email to somebody starting with `b`, i.e. `To: b<tab>` we are being presented with a list of three entries, where we select the first two again, but leave the cursor on Bob Cramer. 

    q:Exit  d:Del  u:Undel  o:Sort  O:Rev-Sort  <Enter>:Select  ?:Help
      1  * beatrice        Beatrice Williams <bw@example.com>                       |
      2  * beatrix         Beatrix Jones <bj@example.com>                           |
      3    bob             Bob Cramer <bc@example.com>                              |

Contrary to the behavior seen with file attachments, Bob Cramer is the only recipient chosen, even though the two ladies were selected:

    To: Bob Cramer <bc@example.com>

## How often does this happen?

- Always

## When did it start to happen?

I am using NeoMutt as of commit 7edc165d4d0de9dc30af706536d8fffdada8f7cf. The selection mechanism stopped working with release tag `20220429`. It worked in NeoMutt release tag `20211029`. Both scenarios described above chose the two ladies and put them into the `To:` field for a binary compiled from source code tag `20211029`. For the current HEAD binary and the binary compiled by checking out tag `20220429` the behavior is as described above.

## Extra Info

* Operating System and its version - _Mac OSX 10.12.6 with Fink installed._
* Were you using multiple copies of NeoMutt at once? - _No._
* Were you using 'screen' or 'tmux'? - _No._
* Is your email local (maildir) or remote (IMAP)? - _Local maildir, but should not apply here._
* NeoMutt is compiled with gcc9 and I can supply more information on the environment and options if needed. 
* Please notice that the query dialog for file attachments works as expected in all binaries! Multiple files can be selected and even if the cursor is not on a tagged file, pressing `<enter>` will attach all selected files.

**Comment by flatcap** [comment]
Ah!
Are you using <kbd>;</kbd><kbd>\<enter\></kbd> (`<tag-prefix><select-entry>`) ?

The behaviour of the Alias / Query dialogs was changed recently to be consistent with other dialogs.

**Comment by flatcap** [comment]
Diagrams of new workflow: https://github.com/neomutt/gfx/blob/main/arch/README.md#alias--query---dialog--completion

**Comment by the-x-at** [comment]
Not to the best of my knowledge. The config file `minimal.muttrc` is the only change from default NeoMutt. This is only to get the example running and reproducible. I am using `<tab>` key to enter the query menu and `t` key to select/tag entries in the presented list. Then I hit `<enter>` to get back to the `To:` entry field.
I however see that with the two tagged messages in the query menu, hitting `;<enter>` will put both recipients into the entry field in the current version of NeoMutt. If this is desired, then the file selection dialog is still working both the old way and the "new way" with `;<enter>`, which was confusing me.
Bottom line: are you saying that `;<enter>` is now the way to select multiple entries from any tagged list? Feature, not a bug? If I translate this series of keystrokes to spoken language, "the next command is using `<enter>` key on all tagged entries", it would even make sense!

**Comment by flatcap** [comment]
> Feature, not a bug?

Yes.  It's a conscious change from Mutt's behaviour.

> it would even make sense!

That's the idea :-)

**Comment by the-x-at** [comment]
Much ado about nothing then. Maybe you want to change the file selection dialog also to only `;<enter>`-select the tagged files?

**Comment by flatcap** [comment]
> Maybe you want to change the file selection dialog also to only `;<enter>`-select the tagged files?

Yes.  I've made a note.
The file browser is a scary bit of code, so it may be a while, though.
